### PR TITLE
CORE-8974 Mark input states of UTXO transaction as consumed with bulk update

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Marking input states of UTXO transaction as consumed with bulk update

### CODE REVIEW

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt [hunk 1]
```diff
@@ -142,21 +143,17 @@ class UtxoRepositoryImpl(
...
     ) {
         entityManager.createNativeQuery(
         """
             UPDATE {h-schema}utxo_relevant_transaction_state
             SET consumed = true
-            WHERE transaction_id = :transactionId
-            AND group_idx = :groupIndex
-            AND leaf_idx = :leafIndex"""
+            WHERE transaction_id in (:transactionIds)
+            AND (transaction_id || ':' || leaf_idx) IN (:stateRefs)"""
```
Comment (vlajos) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:153]: I ended up using the same string manipulation here: https://github.com/corda/corda-runtime-os/pull/2850
I could not find anything neater, but it does not look 100% neat.

Comment (mbrkic-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:153]: Yes, I did in the same way as you. I spent lots of time yesterday trying to do this properly, but seems there is no way unless we create Hibernate user type, which would make code explicitly dependent on Hibernate: https://stackoverflow.com/questions/55317347/jpql-and-list-of-tuples-as-parameter-for-select-in-statements/55431645

**Review by vlajos — APPROVED** [review]

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt (hunk 1 above)
Comment (mnesbit) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:153]: This is part of why consume needs to become a calculated value! The set of staterefs you can pass as an in parameter is often very limited and expensive. So this probably wouldn't work for transactions with many (e.g. 100) inputs

Comment (lankydan) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoRepositoryImpl.kt:153]: So are we saying there is a better way to do this? Do we merge this as is and address later or is it better to deal with it now? It sounds like we might as well address it now?

**Pull request merged**